### PR TITLE
metrics.js -> metrics.ts

### PR DESCRIPTION
**Pull request by denar90**
@samccone if you don't mind. 
This is small one :)

🍰 this time 😛

### CODE REVIEW

**Comment by denar90** [comment]
@pedro93 if you are ok with this one => 💥 

@samccone if here smth to improve I'll make another PR then.
👯 🍺  from me :)

File: lib/metrics.ts [hunk 1]
```diff
@@ -26,50 +27,45 @@ module.exports = {
...
   checkAudits(audits);
 
   const colorP0 = 'yellow';
   const colorP2 = 'green';
   const colorVisual = 'blue';
 
-  const timings = [];
-  const navStart = metricsDefinitions.find(def => def.id === metricsIds.NAVSTART);
-  const timestamps = [{
+  const timings: Array<Timings> = [];
```
Comment (samccone) [inline lib/metrics.ts]: for simple array types prefer Timings[]

File: lib/metrics.ts [hunk 2]
```diff
@@ -26,50 +27,45 @@ module.exports = {
...
+  const navStart = metricsDefinitions.find((def: MetricsDefinitions) => def.id === metricsIds.NAVSTART);
```
Comment (samccone) [inline lib/metrics.ts]: def: type should be inferred

File: lib/metrics.ts [hunk 3]
```diff
@@ -26,50 +27,45 @@ module.exports = {
...
     id: navStart.id,
     timestamp: navStart.getTs(audits)
   }];
 
   metricsDefinitions
-    .filter(def => def.id !== metricsIds.NAVSTART)
-    .forEach(metric => {
-      const resolvedMetric = {
+    .filter((def: MetricsDefinitions) => def.id !== metricsIds.NAVSTART)
+    .forEach((metric: MetricsDefinitions) => {
```
Comment (samccone) [inline lib/metrics.ts]: again, should not be needed here :)

File: types/types.ts [hunk 4]
```diff
@@ -1,6 +1,19 @@
 // Copyright 2016 Google Inc. All Rights Reserved.
 // Licensed under the Apache License, Version 2.0. See LICENSE
 
+//@todo remove stuff after https://github.com/GoogleChrome/lighthouse/pull/1882 be merged and version released
```
Comment (samccone) [inline types/types.ts]: looks like this has landed

File: types/types.ts [hunk 5]
```diff
@@ -19,6 +32,12 @@ interface MetricsResults {
   initialUrl: string;
 }
 
+interface MetricsDefinitions {
+  name: string;
+  id: string;
+  getTs(audits: any): number;
```
Comment (samccone) [inline types/types.ts]: is this really an unknown type? or can we type the any :)

File: types/types.ts (hunk 4 above)
Comment (pedro93) [inline types/types.ts]: Has it been released to npm or is it in the master branch?

Comment (denar90) [inline types/types.ts]: Yeap, we should wait a bit for 1.6.4, 1.6.3 doesn't have it...

File: lib/metrics.ts [hunk 6]
```diff
@@ -26,50 +28,45 @@ module.exports = {
...
+const checkAudits = (audits: LighthouseAudits) => Object.keys(audits).forEach(key => {
   const debugString = audits[key].debugString;
   if (audits[key].debugString)
     throw new Error(`${debugString} Audit key: ${key}`);
 });
 
-function prepareData(res) {
-  const audits = res.audits;
+function prepareData(res: LighthouseResults): MetricsResults {
+  const audits: LighthouseAudits = res.audits;
```
Comment (samccone) [inline lib/metrics.ts]: this should be inferred

**Comment by paulirish** [comment]
sg. lets land after the nit is addressed.

**Comment by denar90** [comment]
@samccone @paulirish thanks 🎉 
I think one last big refactoring PR will be landed for `index.js` in a few days. Count on your review if you don't mind :)

**Pull request merged**